proto/kuksa/val/v2/val.proto

```diff
@@ -0,0 +1,227 @@
+  //    map<SignalID, Error>
+  // }
+  rpc GetValues(GetValuesRequest) returns (GetValuesResponse);
+
+  // List values of signals matching the request.
+  //
+  // Returns a list of signal values. Only values of signals that the user
+  // is allowed to read are included (everything else is ignored).
+  //
+  // Returns (grpc error code):
+  //   NOT_FOUND if the specified root branch does not exist.
+  rpc ListValues(ListValuesRequest) returns (ListValuesResponse);
+
+  // Subscribe to a set of signals
+  // Returns (GRPC error code):
+  //   NOT_FOUND if any of the signals are non-existant.
+  //   PERMISSION_DENIED if access is denied for any of the signals.
+  rpc Subscribe(SubscribeRequest) returns (stream SubscribeResponse);
+
+  // Actuate a single actuator
+  //
+  // Returns (GRPC error code):
+  //   NOT_FOUND if the actuator does not exist.
+  //   PERMISSION_DENIED if access is denied for of the actuator.
+  //   UNAVAILABLE if there is no provider currently providing the actuator
+  //   INVALID_ARGUMENT
+  //       - if the data type used in the request does not match
+  //            the data type of the addressed signal
+  //       - if the requested value is not accepted,
+  //            e.g. if sending an unsupported enum value
+  rpc Actuate(ActuateRequest) returns (ActuateResponse);
+
+  // Actuate simultaneously multiple actuators.
+  // If any error occurs, the entire operation will be aborted
+  // and no single actuator value will be forwarded to the provider.
+  //
+  // Returns (GRPC error code):
+  //   NOT_FOUND if any of the actuators are non-existant.
+  //   PERMISSION_DENIED if access is denied for any of the actuators.
+  //   UNAVAILABLE if there is no provider currently providing an actuator
+  //   INVALID_ARGUMENT
+  //       - if the data type used in the request does not match
+  //            the data type of the addressed signal
+  //       - if the requested value is not accepted,
+  //            e.g. if sending an unsupported enum value
+  rpc BatchActuate(BatchActuateRequest) returns (BatchActuateResponse);
+
+  // List metadata of signals matching the request.
+  //   NOT_FOUND if the specified root branch does not exist.
+  rpc ListMetadata(ListMetadataRequest) returns (ListMetadataResponse);
+
+  // Publish a signal value. Used for low frequency signals (e.g. attributes).
+  //
+  // Returns (GRPC error code):
+  //   NOT_FOUND if any of the signals are non-existant.
+  //   PERMISSION_DENIED
+  //       - if access is denied for any of the signals.
+  //       - if the signal is already provided by another provider.
+  //   INVALID_ARGUMENT
+  //       - if the data type used in the request does not match
+  //            the data type of the addressed signal
+  //       - if the published value is not accepted,
+  //            e.g. if sending an unsupported enum value
+  rpc PublishValue(PublishValueRequest) returns (PublishValueResponse);
```

Comment:
I would like to have a "oneshot" publish for multiple values: With that a provider can easily publish a whole bunch of attributes (I'd prefer the "batch" version over the single one on the provider side)